compaction/size_tiered_compaction_strategy.cc

```diff
@@ -38,8 +38,10 @@ size_tiered_compaction_strategy_options::size_tiered_compaction_strategy_options
     cold_reads_to_omit = DEFAULT_COLD_READS_TO_OMIT;
 }
 
+template <std::ranges::range Range>
+requires std::same_as<std::ranges::range_value_t<Range>, sstables::shared_sstable>
 std::vector<std::pair<sstables::shared_sstable, uint64_t>>
-size_tiered_compaction_strategy::create_sstable_and_length_pairs(const std::vector<sstables::shared_sstable>& sstables) {
+size_tiered_compaction_strategy::create_sstable_and_length_pairs(const Range& sstables) {
```

Comment:
What's the motivation? In general it's bad to over-template things.

Reply:
The motivation is to be able to accept a reference to either a vector or an unordered set

Reply:
Well of course the motivation is to support multiple types, the question is are these necessary.

We should be careful sprinkling templates everywhere in high-level code.